scan/src/components/UserAccount.re

```diff
@@ -179,30 +179,35 @@ module Balance = {
 
 [@react.component]
 let make = () => {
+  let metadataSub = MetadataSub.use();
   let (accountOpt, dispatchAccount) = React.useContext(AccountContext.context);
   let (_, dispatchModal) = React.useContext(ModalContext.context);
 
-  let connect = () => dispatchModal(OpenModal(Connect));
+  let connect = chainID => dispatchModal(OpenModal(Connect(chainID)));
   let disconnect = () => dispatchAccount(Disconnect);
   let submitTx = () => dispatchModal(OpenModal(SubmitTx));
 
-  switch (accountOpt) {
-  | Some({address}) =>
-    <>
-      <Row justify=Row.Right>
-        <Col> <AddressRender address position=AddressRender.Nav /> </Col>
-        <Col> <HSpacing size={`px(27)} /> </Col>
-        <Col> <DisconnectBtn disconnect /> </Col>
-      </Row>
-      <VSpacing size=Spacing.md />
-      <Row justify=Row.Right>
-        <Col> <Balance address /> </Col>
-        <Col> <HSpacing size={`px(5)} /> </Col>
-        <Col> <FaucetBtn address={address->Address.toBech32} /> </Col>
-      </Row>
-      <VSpacing size=Spacing.md />
-      <Row justify=Row.Right> <Col> <SubmitTxBtn submitTx /> </Col> </Row>
-    </>
-  | None => <Col> <ConnectBtn connect /> </Col>
+  switch (metadataSub) {
+  | Data({chainID}) =>
+    switch (accountOpt) {
+    | Some({address}) =>
+      <>
+        <Row justify=Row.Right>
+          <Col> <AddressRender address position=AddressRender.Nav /> </Col>
+          <Col> <HSpacing size={`px(27)} /> </Col>
+          <Col> <DisconnectBtn disconnect /> </Col>
+        </Row>
+        <VSpacing size=Spacing.md />
+        <Row justify=Row.Right>
+          <Col> <Balance address /> </Col>
+          <Col> <HSpacing size={`px(5)} /> </Col>
+          <Col> <FaucetBtn address={address->Address.toBech32} /> </Col>
+        </Row>
+        <VSpacing size=Spacing.md />
+        <Row justify=Row.Right> <Col> <SubmitTxBtn submitTx /> </Col> </Row>
+      </>
+    | None => <Col> <ConnectBtn connect={_ => connect(chainID)} /> </Col>
+    }
+  | _ => React.null
```

Comment:
Can we do the loading animation in this PR? How hard is it?

Reply:
Sure, I'll add it.